example/lib/peripheral_details/peripheral_details_bloc.dart

```diff
@@ -0,0 +1,28 @@
+import 'dart:async';
+import 'package:blemulator_example/adapter/ble_adapter.dart';
+import 'package:blemulator_example/repository/peripheral_repository.dart';
+import 'package:bloc/bloc.dart';
+import './bloc.dart';
+
+class PeripheralDetailsBloc
+    extends Bloc<PeripheralDetailsEvent, PeripheralDetailsState> {
+  BleAdapter _bleAdapter;
+  BlePeripheralRepository _blePeripheralRepository;
+
+  PeripheralDetailsBloc(this._bleAdapter, this._blePeripheralRepository);
+
+  @override
+  PeripheralDetailsState get initialState =>
+      PeripheralDetailsState(peripheral: _blePeripheralRepository.blePeripheral);
+
+  @override
+  Stream<PeripheralDetailsState> mapEventToState(
+    PeripheralDetailsEvent event,
+  ) async* {}
+
+  @override
+  Future<void> close() {
+    _blePeripheralRepository.blePeripheral = null;
```

Comment:
this way, the internal implementation of the repository leaks. Bloc or any other class than repository should not know how to clear picked peripheral.

Reply:
Implementation was changed and BlePeripheralRepository no longer exists in the project.